### PR TITLE
Updated MongoDB driver to 4.8

### PR DESCRIPTION
**Pull request by radekmie**
In this pull request, I updated the MongoDB driver to version 4.8. As mentioned in [here](https://github.com/meteor/meteor/discussions/12092#discussioncomment-3198339), the `connect` operation is no longer mandatory, making it possible to make all asynchronous constructors listed in https://github.com/meteor/meteor/pull/12028 synchronous again.

The only problem right now is that the `MongoConnection` no longer throws an error (nor rejects) when there's a connection problem, making [this test](https://github.com/radekmie/meteor/blob/99e93bde4168a4c287fe371908dc02aae2fbf33d/packages/mongo/mongo_livedata_tests.js#L3232-L3239) fail. I see a few options to handle that:
1. **Ignore the error completely.** That means, we silently ignore it, making all of the subsequent operations on the connection fail. It's not a huge problem, as the connection may break at anytime anyway, and all errors have to be handled properly.
2. **Expose `.connectionPromise`.** That means, we actually call `.connect`, won't wait for the result - simply store the promise on an instance. This way everyone can `await` it if needed.
    * This is doable in the userspace, as [it's fine to call `.connect` multiple times](https://github.com/mongodb/node-mongodb-native/blob/be34a94651c2b18df303f248e10b1bfa06dc445e/src/operations/connect.ts#L17-L20).
3. Something else...?

**EDIT 2022-07-28:** I chose 1. for now.

### CODE REVIEW

File: packages/mongo/collection.js [hunk 1]
```diff
@@ -907,6 +899,6 @@ ASYNC_COLLECTION_METHODS.forEach(methodName => {
 if (Meteor.isServer) {
   const userOptions = Meteor.settings?.packages?.mongo || {};
   if (!userOptions?.skipStartupConnection && !process.env.METEOR_TEST_FAKE_MONGOD_CONTROL_PORT) {
-    Promise.await(MongoInternals.defaultRemoteCollectionDriver());
+    MongoInternals.defaultRemoteCollectionDriver();
   }
```
[the remark this answers is not in the document]
Comment (radekmie) [reply packages/mongo/collection.js:903]: We definitely can, it's not really helpful at the moment. Especially if we ignore the connection error.

**Comment by zodern** [comment]
Many deployment tools, including Meteor Up, send requests to the app during a deploy or when updating its configuration to check if and when the app successfully starts. It would be nice if these could still detect when there is a problem caused by the Mongo connection (maybe the MONGO_URL was changed, or Meteor was updated to a version incompatible with the mongo version, etc.) so they can roll back the changes. For this to happen, Meteor would need to crash before the startup callbacks finish running, since webapp starts listening for incoming http requests right after that.

An overly simple way of dong this might be something like:
```
MongoInternals.defaultRemoteCollectionDriver = _.once(function () {
    // ....
    
    Meteor.startup(() => {
      connectionPromise.await();
    });
    
   return driver;
 });
```

**Comment by fredmaiaarantes** [comment]
> Is the CI flaky? I mean, it ran twice, and two different tests failed.

The failing tests are coming from the `release-2.7.4` branch @radekmie. We should look into them @denihs.

**Comment by radekmie** [comment]
@zodern Sure, I'll add that. Is there a way to test whether the app crashes using `TinyTest`?

**Comment by filipenevola** [comment]
> As mentioned in [here](https://github.com/meteor/meteor/discussions/12092#discussioncomment-3198339), the `connect` operation is no longer mandatory, making it possible to make all asynchronous constructors listed in #12028 synchronous again.

This is great news. My main concern is about MongoDB Driver consistency, if they change this behavior again in the future we would need to go back to the first route but I think this is fine for now.

It [seems](https://github.com/mongodb/node-mongodb-native/releases/tag/v4.7.0) they are committed to keep `connect` optional:
> Note: It's a known limitation that explicit sessions (client.startSession) and initializeOrderedBulkOp, initializeUnorderedBulkOp cannot be used until MongoClient.connect is first called. Look forward to a future patch release that will correct these inconsistencies.

**Review by filipenevola — APPROVED** [review]

**Comment by zodern** [comment]
> Is there a way to test whether the app crashes using TinyTest?

You could maybe use `process.once('uncaughtException')` in the test.

**Comment by radekmie** [comment]
@zodern Cool, will try that tomorrow.

**Pull request merged**